File: 1033-moving-stones-until-consecutive/1033-moving-stones-until-consecutive.go
func numMovesStones(a int, b int, c int) []int {
    x := min(a, min(b, c))
    z := max(a, max(b, c))
    y := a+b+c-x-z
    mi, ma := 0, 0
    if z-x > 2 {
		mi = 2
		if y-x < 3 || z-y < 3 {
			mi = 1
		}
        ma = z-x-2
    }
    return []int{mi, ma}
}

File: 1033-moving-stones-until-consecutive/1033-moving-stones-until-consecutive.js
/**
 * @param {number} a
 * @param {number} b
 * @param {number} c
 * @return {number[]}
 */
var numMovesStones = function(a, b, c) {
    const x = Math.min(a, Math.min(b, c));
    const z = Math.max(a, Math.max(b, c));
    const y = a+b+c-x-z;
    let mi = 0; let ma = 0;
    if(z-x > 2) {
        mi = y-x < 3 || z-y < 3 ? 1 : 2;
        ma = z-x-2;
    }
    return [mi, ma];
};